### PR TITLE
Pin 1323 Implementazione endpoint authorization-process GET /clients/{clientId}

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**

### CODE REVIEW

File: src/main/resources/interface-specification.yml [hunk 1]
```diff
@@ -6329,6 +6329,96 @@ paths:
...
+        - clients
+      summary: Get a Client
+      operationId: getClient
+      parameters:
+        - name: clientId
+          in: path
+          description: The Client id
+          required: true
+          schema:
+            type: string
```
Comment (galales) [inline src/main/resources/interface-specification.yml:6486]:
```suggestion
            type: string
            format: uuid
```

File: src/main/resources/interface-specification.yml [hunk 2]
```diff
@@ -7796,6 +7886,62 @@ components:
           $ref: '#/components/schemas/AgreementApprovalPolicy'
         eservice:
           $ref: '#/components/schemas/CatalogDescriptorEService'
+    ClientDetails:
```
Comment (galales) [inline src/main/resources/interface-specification.yml]:
```suggestion
    Client:
```
After [this](https://github.com/pagopa/interop-be-backend-for-frontend/pull/141#discussion_r1139988911) suggestion has been accepted

File: src/main/resources/interface-specification.yml [hunk 3]
```diff
@@ -7796,6 +7886,62 @@ components:
...
+    Organization:
+      type: object
+      required:
+        - id
+        - description
+      properties:
+        id:
+          type: string
+        description:
+          type: string
```
Comment (galales) [inline src/main/resources/interface-specification.yml]: `CompactOrganization` already exists
```suggestion
```

File: src/main/resources/interface-specification.yml [hunk 4]
```diff
@@ -7796,6 +7886,62 @@ components:
...
           $ref: '#/components/schemas/CatalogDescriptorEService'
+    ClientDetails:
+      description: Models Client details
+      type: object
+      properties:
+        id:
+          type: string
+          format: uuid
+        consumer:
+          $ref: '#/components/schemas/Organization'
```
Comment (galales) [inline src/main/resources/interface-specification.yml]:
```suggestion
          $ref: '#/components/schemas/CompactOrganization'
```

File: src/main/resources/interface-specification.yml [hunk 5]
```diff
@@ -7796,6 +7886,62 @@ components:
...
+    ClientPurpose:
+      type: object
+      properties:
+        purposeId:
+          type: string
+          format: uuid
+        title:
+          type: string
+        eservice:
+          $ref: '#/components/schemas/CompactEServiceLight'
```
Comment (galales) [inline src/main/resources/interface-specification.yml]: let's confirm it with the frontend, but we should maybe use `CompactEService` here (which has the producer field)

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ClientsApiServiceImpl.scala [hunk 6]
```diff
@@ -206,4 +206,22 @@ final case class ClientsApiServiceImpl(authorizationProcessService: Authorizatio
...
+  def getClient(clientId: String)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerClientDetails: ToEntityMarshaller[ClientDetails]
+  ): Route = {
+
+    val result: Future[ClientDetails] = for {
+      clientUuid <- clientId.toFutureUUID
+      client     <- authorizationProcessService.getClient(clientUuid)
+    } yield (client.toApi)
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ClientsApiServiceImpl.scala]:
```suggestion
    } yield client.toApi
```

**Review by TonioGela — APPROVED** [review]

File: src/main/resources/interface-specification.yml (hunk 2 above)
Comment (nttdata-rtorsoli) [reply src/main/resources/interface-specification.yml]: Client already exists, just spoke with Carmine and FE wants back the entire info linked to Client. However I will check it again with FE guys.

File: src/main/scala/it/pagopa/interop/backendforfrontend/service/types/AuthorizationProcessServiceTypes.scala [hunk 7]
```diff
@@ -12,6 +12,36 @@ object AuthorizationProcessServiceTypes {
...
+  }
+
+  implicit class ClientKindConverter(private val ck: AuthorizationProcess.ClientKind) extends AnyVal {
+    def toApi: ClientKind = ck match {
+      case AuthorizationProcess.ClientKind.API      => ClientKind.API
+      case AuthorizationProcess.ClientKind.CONSUMER => ClientKind.CONSUMER
+    }
+  }
+
+  implicit class ClientProcessKindConverter(private val ck: ClientKind) extends AnyVal {
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/backendforfrontend/service/types/AuthorizationProcessServiceTypes.scala]:
```suggestion
  implicit class ClientKindConverter(private val ck: ClientKind) extends AnyVal {
```

**Review by galales — APPROVED** [review]

**Review by galales — APPROVED** [review]

**Pull request merged**